### PR TITLE
Introduce some recursion into getting config options

### PR DESCRIPTION
**Pull request by khalidharun**
Break down get config into a recursive function.  Probably overkill for most cases, which are only 2 levels deep but I thought I'd throw it in here.

### CODE REVIEW

**Comment by peterhurford** [comment]
Looks good. Can you add a quick test for your recursion? :+1:

**Comment by khalidharun** [comment]
@peterhurford ready for review

**Comment by peterhurford** [comment]
@khalidharun Last request -- can you version and add to `NEWS.md`?

**Comment by khalidharun** [comment]
@peterhurford done and done.

File: R/utils.R [hunk 1]
```diff
@@ -4,3 +4,16 @@
...
+#' Helper function to get settings from a nested config file
+#'
+#' @param config list. The output from a YML file as a list.
+#' @param env character. The keys of a nested to list to find the settings.
+#' @return The settings if found else `NULL`.
+get_config <- function(config, env) {
+  if (!is.character(env)) stop("env is not a character.")
+  if (!is.list(config)) stop("config is not a list.")
+  if (length(env) > 1) Recall(config[[env[1]]], env[-1])
+  config[[env]]
```
Comment (robertzk) [inline R/utils.R]: @khalidharun What about 

```r
get_config <- function(config, env) {
  if (!is.character(env)) stop("env is not a character.")
  if (!is.list(config)) stop("config is not a list.")
  config[[env]]
}
```

For example,

```r
list(a = list(b = list(c = 1)))[[c("a", "b", "c")]]
```

See `help("[[")`

**Pull request merged**